src/store/workflows.module.js

```diff
@@ -14,9 +14,38 @@
  * You should have received a copy of the GNU General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
+import applyDeltasGScan from '@/components/cylc/gscan/deltas'
+import applyDeltasLookup from '@/components/cylc/workflow/deltas'
+import applyDeltasTree from '@/components/cylc/tree/deltas'
+import Alert from '@/model/Alert.model'
+import { clear } from '@/components/cylc/tree/index'
 
 const state = {
+  /**
+   * @type {Object.<String, Object>}
+   */
+  lookup: {},
+  /**
+   * This is the CylcTree, which contains the hierarchical tree data structure.
+   * It is created from the GraphQL data, with the only difference that this one
+   * contains hierarchy, while the lookup (not workflow.lookup) is flat-ish.
+   *
+   * The nodes in the .tree property have a reference or pointer (.node) to the
+   * data in the lookup map above, to avoid data duplication.
+   *
+   * @type {Workflow}
+   */
+  workflow: {
+    tree: {},
+    lookup: {}
```

Comment:
Ah, mea culpa too, I'm horrible naming things.

The `workflows/lookup` is the workflow view lookup used to look up GraphQL raw data. The idea is for components like Tree & Table to re-use the data in `workflows/lookup` as pointers, instead of duplicating it.

The `workflows/workflow` holds the data for a Tree view. Its`workflows/workflow.tree` contains just a single root node (workflow) that contains children data (hierarchical data).

And `workflows/workflow.lookup` is used as a lookup of tree nodes. When you add a new node to the tree, instead of using the `node.children`, we just grab the parent node from `workflows/workflow.lookup` and add the new node as child.

The new node will have `{ id: bla, node: $POINTER_TO_WORKFLOWS_LOOKUP_DATA, children: [] }`.

Hope this makes sense, and happy to update these names (just don't trust me to choose better names :laughing: )

Reply:
That makes sense, thanks for explaining. I think those names are great, just getting my head around the distinction between the `workflows` and `workflow` levels.

I guess with the hierarchical GScan we would add another `tree` structure at the `workflows` level.

Reply:
>I guess with the hierarchical GScan we would add another tree structure at the workflows level.

Yup, probably a very similar structure (since GScan re-uses the Tree component) :+1: